include/swift/Driver/Action.h

```diff
@@ -321,6 +325,8 @@ class LinkJobAction : public JobAction {
   static bool classof(const Action *A) {
     return A->getKind() == Action::Kind::LinkJob;
   }
+
+  virtual bool allowsResponseFiles() const { return true; }
```

Comment:
I'm pretty sure this is a toolchain-dependent decision, and even without that it doesn't really make sense at the Action level. Action is supposed to be independent of the executable you use to accomplish that action.